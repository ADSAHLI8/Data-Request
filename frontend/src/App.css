body {
    background:  #0891b2;
    background-size: 400% 400%;
    animation: gradient 15s ease infinite;
    font-family: "Roboto", sans-serif;
}



.p-tabview .p-tabview-panels{
    background: none;
}
.p-tabview .p-tabview-nav{
    background: none;
    display: none;
}

.p-tabview .p-tabview-nav li .p-tabview-nav-link{
    background: none;
}
.full-container{
    margin: 5% 25%;
    padding: 2%;
    padding-bottom: 0%;
    padding-top: 0%;
    background-color: #f2f2f2d6;
    border-radius: 20px;
}
div.p-card:nth-child(1) > div:nth-child(1){
    background: none;
    margin-bottom: 1%;
    border-radius: 8px;
    padding-bottom: 0%;
}
div.p-card:nth-child(1){
    border-radius: 8px;
    background: none;
}
div.p-card:nth-child(2) > div:nth-child(1){
    background: none;
    margin-bottom: 1%;
    border-radius: 8px;
    padding-bottom: 0%;

}
div.p-card:nth-child(2){
    background: none;
    border-radius: 8px;
}
div.p-card:nth-child(2) > div:nth-child(1) > div:nth-child(2){
    padding: 1%;
    padding-left: 0%;
}
.p-tabview .p-tabview-panels{
    padding-top: 0%;
}