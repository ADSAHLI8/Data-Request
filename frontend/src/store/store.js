import { configureStore } from '@reduxjs/toolkit'
import DataRequestReducer from '../features/DataRequest/dataRequestSlice'
import InsertionConditionReducer from '../features/InsertionCondition/insertionConditionSlice'
import {requestApi} from '../features/requestApi'


 const store = configureStore({
    reducer: {
        dataRequest : DataRequestReducer,
        insertionCondition : InsertionConditionReducer,
        [requestApi.reducerPath] : requestApi.reducer
    },
    middleware : (getDefaultMiddleware) => getDefaultMiddleware().concat(requestApi.middleware)
    
})
export default store