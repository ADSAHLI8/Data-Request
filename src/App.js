import React from 'react';
import { PrimeReactProvider } from 'primereact/api';
import { TabView, TabPanel } from 'primereact/tabview';import "primereact/resources/themes/lara-light-cyan/theme.css";
import './App.css';
import DataRequest from './features/DataRequest/DataRequest';
import InsertionCondition from './features/InsertionCondition/InsertionCondition';
import { Accordion, AccordionTab } from 'primereact/accordion';
import 'primeicons/primeicons.css';
import Submit from './features/submit/Submit';

/**
 * @component
 * 
 * @returns{JSX.Element} 
 */



function App() {
  
  
  return (
    /**
     * Prime React Provider
     * Components
     */
     <PrimeReactProvider>
      <div className='full-container'>
        <TabView>
          <TabPanel >
            <Accordion activeIndex={0}>
            <AccordionTab header="Data Request">
            <DataRequest/>
            </AccordionTab>
             <AccordionTab header="Insertion Condition">
             <InsertionCondition/>
             </AccordionTab>
             </Accordion>
         <Submit/>
        </TabPanel>
      </TabView>
      </div>

     </PrimeReactProvider>

  );
}

export default App;
