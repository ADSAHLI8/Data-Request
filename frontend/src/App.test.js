import { Provider } from "react-redux";
import { render,screen,fireEvent,waitFor,within } from "@testing-library/react";
import '@testing-library/jest-dom'; // For matchers like 'toBeInTheDocument'
import store from "./store/store";
import renderer from 'react-test-renderer';
import App from "./App";

const MockedApp = () => {
    return(
        <Provider store={store}>
              <App />
        </Provider>
    )
}
describe('App', ()=> {
    it('should render accordions properly',async ()=>{
        render(<MockedApp />)
        const buttonElement = screen.getByRole('button',{name : 'Data Request'})
        const buttonElement2 = screen.getByRole('button',{name : 'Insertion Condition'})
        expect(buttonElement).toBeInTheDocument()
        expect(buttonElement2).toBeInTheDocument()
        fireEvent.click(buttonElement)
        const dataRequestElement = screen.getByRole('button',{name:'itm'})
        expect(dataRequestElement).toBeInTheDocument()
        fireEvent.click(buttonElement2)
        await waitFor(() => {
            const Element2 = screen.getByRole('button',{name:'Enabled'})
            expect(Element2).toBeInTheDocument()
          })
    })
    it('should render correctly',async () => {
        const component = renderer.create(<MockedApp/>);
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('Should display buttons values on request message',async () =>{
        render(<MockedApp />)
        const buttonElement=screen.getByRole('button',{name : /itm/i})
        fireEvent.click(buttonElement)
        const buttonElement2 = screen.getByRole('button',{name : /EUROPE/i})
        fireEvent.click(buttonElement2)
        const buttonElement3 = screen.getByRole('button',{name : 'Insertion Condition'})
        fireEvent.click(buttonElement3)
        const buttonElement4 = screen.getByRole('button',{name : 'Enabled'})
        fireEvent.click(buttonElement4)

        
        const submitButton = screen.getByRole('button',{name : 'Submit'})
        fireEvent.click(submitButton)
        await screen.findByRole('dialog')
        const modal = screen.getByRole('dialog')   
        
        expect(within(modal).getByText(/itm/i)).toBeInTheDocument()
        expect(within(modal).getByText(/EUROPE/i)).toBeInTheDocument()
        expect(within(modal).getByText(/Enabled/i)).toBeInTheDocument()

    })
    it('Should display inputs values on request message',async () =>{
        render(<MockedApp />)
        const inputElement=screen.getByPlaceholderText('Input Domain')
        fireEvent.change(inputElement,{target:{value:'domdom'}})
        const inputElement1=screen.getByPlaceholderText('Input Dashboard Name')
        fireEvent.change(inputElement1,{target:{value:'dashdash'}})
        const buttonElement = screen.getByRole('button',{name : 'Insertion Condition'})
        fireEvent.click(buttonElement)
        const inputElement2 = screen.getByPlaceholderText('Input Provider')
        fireEvent.change(inputElement2,{target:{value:'provider2'}})
        const inputElement3 = screen.getByPlaceholderText('Input ISP')
        fireEvent.change(inputElement3,{target:{value:'isp2'}})
        const inputElement4 = screen.getByPlaceholderText('Input List Name')
        fireEvent.change(inputElement4,{target:{value:'list-list'}})
        const textAreaElement = screen.getByPlaceholderText('Any additional information...')
        fireEvent.change(textAreaElement,{target:{value:'additional info ...'}})

        
        const submitButton = screen.getByRole('button',{name : 'Submit'})
        fireEvent.click(submitButton)
        await screen.findByRole('dialog')
        const modal = screen.getByRole('dialog')
        
        expect(within(modal).getByText(/domdom/i)).toBeInTheDocument()
        expect(within(modal).getByText(/dashdash/i)).toBeInTheDocument()
        expect(within(modal).getByText(/provider2/i)).toBeInTheDocument()
        expect(within(modal).getByText(/isp2/i)).toBeInTheDocument() 
        expect(within(modal).getByText(/additional info.../i)).toBeInTheDocument()
    })
    it('Should display select values on request message',async () =>{
        render(<MockedApp />)
        const selectElement=screen.getByText('Select ISP')
        fireEvent.mouseDown(selectElement)
        const option = screen.getByText('orange')
        expect(option).toBeInTheDocument()
        fireEvent.click(option)

        const selectElement2=screen.getByText('Select Another Country')
        fireEvent.mouseDown(selectElement2)
        const option2 = screen.getByText('AF')
        expect(option2).toBeInTheDocument()  
        fireEvent.click(option2)

        const submitButton = screen.getByRole('button',{name : 'Submit'})
        fireEvent.click(submitButton)
        await screen.findByRole('dialog')
        const modal = screen.getByRole('dialog')
        
        expect(within(modal).getByText(/orange/i)).toBeInTheDocument() 
        expect(within(modal).getByText(/af/i)).toBeInTheDocument() 
    })
    it('Should display switch values on request message',async () =>{
        render(<MockedApp />)
        const switchElements=screen.getAllByRole('switch')
        const switchElement = switchElements[0]
        fireEvent.click(switchElement)
    
        const submitButton = screen.getByRole('button',{name : 'Submit'})
        fireEvent.click(submitButton)
        await screen.findByRole('dialog')
        const modal = screen.getByRole('dialog')
        
        expect(within(modal).getByText(/is_Amazon/i)).toBeInTheDocument() 
    })
    it('should reset form',()=>{
        render(<MockedApp />)
        const inputElement=screen.getByPlaceholderText('Input Domain')
        fireEvent.change(inputElement,{target:{value:'domdom'}})
        const resetButton = screen.getByRole('button',{name : /reset/i})
        fireEvent.click(resetButton)

        expect(screen.queryByText(/domdom/i)).not.toBeInTheDocument()

    })
})
