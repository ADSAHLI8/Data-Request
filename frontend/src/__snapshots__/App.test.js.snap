// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="full-container"
>
  <div
    className="p-tabview p-component"
    data-pc-name="tabview"
    data-pc-section="root"
    id={null}
    style={Object {}}
  >
    <div
      className="p-tabview-nav-container"
      data-pc-section="navcontainer"
    >
      <div
        className="p-tabview-nav-content"
        data-pc-section="navcontent"
        id="null_navcontent"
        onScroll={[Function]}
        style={Object {}}
      >
        <ul
          className="p-tabview-nav"
          data-pc-section="nav"
          role="tablist"
        >
          <li
            className="p-unselectable-text p-tabview-selected p-highlight"
            data-pc-name=""
            data-pc-section="header"
            role="presentation"
            style={Object {}}
          >
            <a
              aria-controls="0_content"
              aria-disabled={false}
              aria-selected={true}
              className="p-tabview-nav-link"
              data-pc-section="headeraction"
              id="null_header_0"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="tab"
              tabIndex={0}
            >
              <span
                className="p-tabview-title"
                data-pc-section="headertitle"
              />
            </a>
          </li>
          <li
            aria-hidden="true"
            className="p-tabview-ink-bar"
            data-pc-section="inkbar"
            role="presentation"
          />
        </ul>
      </div>
    </div>
    <div
      className="p-tabview-panels"
      data-pc-section="panelcontainer"
      style={Object {}}
    >
      <div
        aria-labelledby="null_header_0"
        className="p-tabview-panel"
        data-pc-name=""
        data-pc-section="content"
        id="0_content"
        role="tabpanel"
        style={Object {}}
      >
        <div>
          <button
            aria-label="Reset"
            className="buttons p-button p-component"
            data-pc-name="button"
            data-pc-section="root"
            disabled={false}
            onClick={[Function]}
          >
            <span
              className="p-button-label p-c"
              data-pc-section="label"
            >
              Reset
            </span>
          </button>
          <button
            aria-label="Submit"
            className="buttons p-button p-component"
            data-pc-name="button"
            data-pc-section="root"
            disabled={false}
            onClick={[Function]}
          >
            <span
              className="p-button-label p-c"
              data-pc-section="label"
            >
              Submit
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
